extensions/amazon-aws/src/actions/logs.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  CloudWatchLogsClient,
+  DescribeLogStreamsCommand,
+  FilteredLogEvent,
+  FilterLogEventsCommand,
+  LogStream,
+} from "@aws-sdk/client-cloudwatch-logs";
+import { AWS_URL_BASE } from "../constants";
+import { LogStartTimes } from "../interfaces";
+
+const cloudWatchLogsClient = new CloudWatchLogsClient({});
+export async function fetchLogs(
+  logGroupName: string,
+  startTime: LogStartTimes,
+  logStreamNamePrefix?: string,
+  logStreamNames?: string[]
+): Promise<FilteredLogEvent[]> {
+  if (!process.env.AWS_PROFILE) return [];
+  return await fetchAllLogs(logGroupName, startTime, logStreamNamePrefix, logStreamNames);
+}
+
+export async function fetchLogStreams(logGroupName: string): Promise<LogStream[]> {
+  if (!process.env.AWS_PROFILE) return [];
+  return await fetchAllLogStreams(logGroupName);
+}
+
+async function fetchAllLogStreams(logGroupName: string, token?: string, accEvents?: LogStream[]): Promise<LogStream[]> {
+  const { logStreams, nextToken } = await cloudWatchLogsClient.send(
+    new DescribeLogStreamsCommand({ logGroupName, nextToken: token })
+  );
+
+  const combinedEvents = [...(accEvents || []), ...(logStreams || [])];
```

Comment:
I'm running into a "heap out of memory" error when listing log streams. That makes sense to me as there can be tens of thousands of log streams for a given log group. I suggest capping this at e.g. 1000:
```js
if (combinedEvents.length > 1000) return combinedEvents;
```